src/picongpu/include/simulationControl/MovingWindow.hpp

```diff
@@ -51,45 +51,90 @@ class MovingWindow
+
+            const uint32_t gpuLocalSizeY = subGrid.getLocalDomain().size.y();
+            const float_64 cellHeight = float_64(cellSize.y());
+
+            const float_64 deltaWayPerStep = (windowMovingSpeed * float_64(DELTA_T));
+
+            /* How many cells the virtual particle with speed of light is pushed forward
+             * at the begin of the simulation.
+             * The number of cells is round up thus `slide_point` define the minimal space
+             * before the virtual particle and that we not get half cells as offset.
+             */
+            const uint32_t slidePointOffsetCells = math::ceil(
+                float_64(globalWindowSizeY) * (float_64(1.0) - slide_point)
+            );
+
+            /* Is the point when the virtual particle **passed** the GPU next to the last
+             * in the current step
+             */
+            const uint32_t firstSlideStep = math::floor(
+                float_64(subGrid.getGlobalDomain().size.y() - slidePointOffsetCells) * cellHeight / deltaWayPerStep
+            );
+
+            /* size of the moving window in pic length */
+            const float_64 windowRealSizeY = float_64(globalWindowSizeY - slidePointOffsetCells) * cellHeight;
+            /* Is the point when the virtual particle **passed** the moving window
+             * in the current step
+             */
+            const uint32_t firstMoveStep = math::floor(
+                windowRealSizeY / deltaWayPerStep
+            );
 
-            /* round to nearest cell to have smoother offset jumps */
-            if (offsetFirstGPU)
+            if (firstMoveStep <= currentStep)
             {
-                *offsetFirstGPU = math::floor(((float_64) stepsInLastGPU + stepsInFutureAfterComma) *
-                                              light_way_per_step / cell_height + 0.5);
+                /* calculate the current position of the virtual particle */
+                const float_64 currentVParticlePosition = deltaWayPerStep * float_64(currentStep);
+                const float_64 currentVParticleCell = currentVParticlePosition / cellHeight;
+                const uint32_t virtualParticlePosition = uint32_t(math::floor(currentVParticleCell)) + slidePointOffsetCells;
+
+                /* calculate the position of the virtual particle after the current step is calculated */
+                const float_64 nextVParticlePosition = deltaWayPerStep * float_64(currentStep + 1);
+                const float_64 nextVParticleCell = nextVParticlePosition / cellHeight;
+                /* This position is used to detect the point in time where the virtual particle
+                 * moves over a GPU border.
+                 */
+                const uint32_t virtualParticleNextPosition =  uint32_t(math::floor(nextVParticleCell)) + slidePointOffsetCells;
+
+                /* Moving window start moving if the virtual particle passed the border of
```

Comment:
below this part of the member we are checking if our *move* does cause a *slide*
-> fix comment to something like

```
slide handling starts here ...
```